OpenRA.Game/Renderer.cs

```diff
@@ -327,17 +327,19 @@ public void EndFrame(IInputHandler inputHandler)
 			renderType = RenderType.None;
 		}
 
-		public void DrawBatch(Vertex[] vertices, int numVertices, PrimitiveType type)
+		public void DrawBatch(IShader shader, Vertex[] vertices, int numVertices, PrimitiveType type)
 		{
 			tempBuffer.SetData(vertices, numVertices);
-			DrawBatch(tempBuffer, 0, numVertices, type);
+			DrawBatch(shader, tempBuffer, 0, numVertices, type);
 		}
 
-		public void DrawBatch<T>(IVertexBuffer<T> vertices,
+		public void DrawBatch(IShader shader, IVertexBuffer vertices,
 			int firstVertex, int numVertices, PrimitiveType type)
-			where T : struct
 		{
 			vertices.Bind();
+
+			// Future notice: using ARB_vertex_array_object makes all the gl calls in LayoutAttributes obsolete.
```

Comment:
I suggest having it as a separate commit so that we can see how it would look and discuss any implementation details, but then defer it to a future PR so that we can merge this one for the next release as the last supporting `GLProfile.Legacy`.